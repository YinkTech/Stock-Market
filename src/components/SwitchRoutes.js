import React from 'react'
import { NavBar } from './NavBar'
import { App } from './App'
import { Details } from './Details'
import { Route, Routes } from 'react-router-dom'

export const SwitchRoutes = () => {
  return (
    <>
      <NavBar />
        <Routes>
          <Route path='/Stock-Market' element={<App />} exact />
          <Route path='/Details' element={<Details />} />
        </Routes>
    </>
  )
}

