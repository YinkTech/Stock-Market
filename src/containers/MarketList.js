import React, { useEffect } from 'react'
import axios from 'axios';
import { useDispatch } from 'react-redux'
import Market from '../components/Market';
import { setMarket } from '../redux/actions/action';
import Filter from '../components/Filter';

export const MarketList = () => {
  const dispatch = useDispatch();

  const handleFilter = (pageNumber) => {
    return (fetchMarket(pageNumber));
  };


  const fetchMarket = async (pageNumber, pageSize=50) => {
    const response = await axios
    .get(`https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=${pageSize}&page=${pageNumber}&sparkline=false`)
    .catch((err) => {
      console.log("err", err);
    });
    dispatch(setMarket (response.data));
  };
  useEffect(() => {
    fetchMarket();
  }, []);

  return (
      <div className='p-2 text-center'>
        <h4 className='lead-text mb-4'>Coins List </h4>
        <Filter className="mb-2" handleFilter={handleFilter} />
       <Market />
    </div>
  )
}
