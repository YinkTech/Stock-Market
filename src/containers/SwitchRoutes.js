import React from 'react'
import { NavBar } from '../components/NavBar'
import { App } from '../components/App'
import { Links } from './../components/Links';
import { Details } from '../components/Details'
import { Route, Routes } from 'react-router-dom'

export const SwitchRoutes = () => {
  return (
    <>
      <NavBar />
      <div className='d-md-flex'>
      <Links /> 
        <Routes>
          <Route path='/Stock-Market' element={<App />} exact />
          <Route path='/Details/:marketId' element={<Details />} />
          <Route>404 not Found! </Route>
        </Routes>
      </div>
    </>
  )
}

