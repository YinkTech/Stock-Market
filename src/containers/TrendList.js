import React, { useEffect } from 'react'
import axios from 'axios';
import { useDispatch } from 'react-redux'
import { Trends } from '../components/Trends';
import { setTrends } from '../redux/actions/action';

export const TrendList = () => {
  const dispatch = useDispatch();
  const fetchTrends = async () => {
    const response = await axios
    .get("https://api.coingecko.com/api/v3/search/trending")
    .catch((err) => {
      console.log("err", err);
    });
    dispatch(setTrends (response.data.coins));
  };
  useEffect(() => {
    fetchTrends();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
      <div className='p-2 text-center trnds'>
        <h4 className='lead-sec-text mb-4'>Trending Coins List</h4>
       { <Trends /> }
    </div>
  )
}

