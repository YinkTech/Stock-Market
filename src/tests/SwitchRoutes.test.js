import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from './../redux/store/store';
import { SwitchRoutes } from './../containers/SwitchRoutes';

describe('App component', () => {
  test('Snapshot test', () => {
    const app = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <SwitchRoutes />
        </BrowserRouter>
      </Provider>,
    )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});