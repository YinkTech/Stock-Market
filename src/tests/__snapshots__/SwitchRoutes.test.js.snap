// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot test 1`] = `
Array [
  <div
    className="nav justify-content-center sticky-top bg-dark text-decoration-none text-white"
  >
    <div
      className="d-flex align-items-center"
    >
       
      <i
        className="bi bi-cash-coin fs-2 px-3"
      />
       
      <a
        className="logo text-uppercase fs-2"
        href="/Stock-Market"
        onClick={[Function]}
      >
         Stock Market 
      </a>
    </div>
  </div>,
  <div
    className="d-md-flex"
  >
    <div
      className="links d-none d-sm-flex p-2"
      id="my-aside"
    >
      <a
        className="home"
        href="/Stock-Market"
        onClick={[Function]}
      >
        <i
          className="bi bi-house me-1"
        />
        Home
      </a>
    </div>
  </div>,
]
`;
